pygfx/utils/color.py

```diff
@@ -335,11 +376,190 @@ def from_hsl(cls, hue, saturation, lightness):
+        provides perceptually uniform brightness and saturation.
+
+        The alpha channel is optional and defaults to 1.
+        """
+        color = Color.from_hsluv(hue, saturation, lightness)
+        color._val[3] = alpha
+        return color
+
+    def to_hsluv(self):
+        """Get the color represented in the HSLuv colorspace, as 3 floats."""
+        h, s, light = hsluv.rgb_to_hsluv(self.rgb)
+        return h / 360.0, s / 100.0, light / 100.0
+
+    def to_hsluva(self):
+        """Get the color represented in the HSLuv colorspace, as 3 floats."""
+        h, s, light = hsluv.rgb_to_hsluv(self.rgb)
+        return h / 360.0, s / 100.0, light / 100.0, self.a
+
+    def lerp(self, target, t):
+        """Linear interpolate from source color towards target color with factor t.
+
+        Parameters
+        ----------
+        target : Color
+            The target color to interpolate towards
+        t : float
+            Interpolation factor between 0 and 1
+
+        Returns
+        -------
+        Color
+            The interpolated color
+        """
+        r = self.r + (target.r - self.r) * t
+        g = self.g + (target.g - self.g) * t
+        b = self.b + (target.b - self.b) * t
+        a = self.a + (target.a - self.a) * t
+        return Color(r, g, b, a)
+
+    def lerp_in_hue(self, target, t, colorspace="hsl"):
+        """Linear interpolate from source color towards target color with factor t in specified colorspace.
+
+        Parameters
+        ----------
+        target : Color
+            The target color to interpolate towards
+        t : float
+            Interpolation factor between 0 and 1
+        colorspace : str
+            The colorspace to interpolate in. One of "hsl", "hsluv", "hsv"
+
+        Returns
+        -------
+        Color
+            The interpolated color
+        """
+        if colorspace == "hsl":
+            h1, s1, l1 = self.to_hsl()
+            h2, s2, l2 = target.to_hsl()
+            to_rgba = lambda h, s, light, a: Color.from_hsla(h, s, light, a)
+        elif colorspace == "hsluv":
+            h1, s1, l1 = self.to_hsluv()
+            h2, s2, l2 = target.to_hsluv()
+            to_rgba = lambda h, s, light, a: Color.from_hsluva(h, s, light, a)
+        elif colorspace == "hsv":
+            h1, s1, l1 = self.to_hsv()
+            h2, s2, l2 = target.to_hsv()
+            to_rgba = lambda h, s, light, a: Color.from_hsva(h, s, light, a)
+        else:
+            raise ValueError(f"Unknown colorspace {colorspace}")
+
+        # Special case for hue - interpolate along shortest path
+        if abs(h2 - h1) > 0.5:
+            if h1 > h2:
+                h2 += 1.0
+            else:
+                h1 += 1.0
+        h = (h1 + (h2 - h1) * t) % 1.0
+        s = s1 + (s2 - s1) * t
+        light = l1 + (l2 - l1) * t
+        a = self.a + (target.a - self.a) * t
+
+        return to_rgba(h, s, light, a)
+
+    def lighter(self, factor=0.5, colorspace="hsl"):
```

Comment:
Since hsluv is more human friendly, maybe that should be the default? Or is there a compatibility of expectation that makes hsl a better default?

Reply:
HSL is very commonly used I believe

Reply:
I think the more important question should we implement lighter in RGB where we interpolate to "white" and to "black" when it's "darker"? or just stick to the Hue based colorspaces with Lightness components only

Reply:
Qt does the conversion in hsv: https://doc.qt.io/qt-6/qcolor.html#lighter Let's do that too.

Reply:
If we do lightening in HSLuv by default, then we won't need to adjust the saturation, so the updated color won't look oversaturated.

But that requires the HSLuv to be a non-optional dependency

Reply:
updated. made hsluv a default option for interpolation and lighter/darker